lib/block_view_validator.go

```diff
@@ -1671,6 +1824,15 @@ func (bav *UtxoView) IsValidUnregisterAsValidatorMetadata(transactorPublicKey []
 		return errors.Wrapf(RuleErrorValidatorNotFound, "UtxoView.IsValidUnregisterAsValidatorMetadata: ")
 	}
 
+	// Validate BLSPKIDEntry exists
+	blsPKIDEntry, err := bav.GetBLSPKIDEntryByBLSPublicKey(validatorEntry.VotingPublicKey)
+	if err != nil {
+		return errors.Wrap(err, "UtxoView.IsValidUnregisterAsValidatorMetadata: ")
+	}
+	if blsPKIDEntry == nil {
+		return errors.Wrap(RuleErrorValidatorBLSPKIDEntryNotFound, "UtxoView.IsValidUnregisterAsValidatorMetadata: ")
+	}
+
```

Comment:
If the entry doesn't exist, do we want to error? Granted this edge case should never happen, but I'm wondering what the safe way to handle it is

Reply:
I think a RuleError here is okay. We got in a bad state if this happened so we probably want to toss this view/txn?

Reply:
Sounds good to me.